packages/app/features/settings/PrivacyScreen.tsx

```diff
@@ -0,0 +1,117 @@
+import { NativeStackScreenProps } from '@react-navigation/native-stack';
+import * as db from '@tloncorp/shared/db';
+import { Text } from '@tloncorp/ui';
+import { useCallback, useState } from 'react';
+import { Switch } from 'react-native';
+import { YStack } from 'tamagui';
+
+import { useTelemetry } from '../../hooks/useTelemetry';
+import { RootStackParamList } from '../../navigation/types';
+import {
+  ScreenHeader,
+  SizableText,
+  View,
+  XStack,
+  triggerHaptic,
+  useStore,
+} from '../../ui';
+
+type Props = NativeStackScreenProps<RootStackParamList, 'PrivacySettings'>;
+
+interface PrivacyState {
+  telemetryDisabled: boolean;
+  phoneDiscoverable: boolean;
+}
+
+export function PrivacySettingsScreen(props: Props) {
+  const store = useStore();
+  const phoneAttest = store.useCurrentUserPhoneAttestation();
+  const telemetry = useTelemetry();
+
+  const [state, setState] = useState<PrivacyState>({
+    phoneDiscoverable: parsePhoneDiscoverability(phoneAttest),
+    telemetryDisabled: telemetry.getIsOptedOut(),
+  });
+
+  const togglePhoneDiscoverable = useCallback(async () => {
+    if (!phoneAttest) {
+      return;
+    }
+    const nextDiscoveryState = !state.phoneDiscoverable;
+    const nextDiscoveryValue: db.AttestationDiscoverability = nextDiscoveryState
+      ? 'verified'
+      : 'hidden';
+    setState((prev) => ({ ...prev, phoneDiscoverable: nextDiscoveryState }));
+    try {
+      await store.updateAttestationDiscoverability({
+        attestation: phoneAttest,
+        discoverability: nextDiscoveryValue,
+      });
+    } catch (e) {
+      triggerHaptic('error');
+      setState((prev) => ({ ...prev, phoneDiscoverable: !nextDiscoveryState }));
+    }
+  }, [phoneAttest, state.phoneDiscoverable, store]);
+
+  const toggleSetTelemetry = useCallback(() => {
+    const nextDisabledState = !state.telemetryDisabled;
+    setState((prev) => ({ ...prev, telemetryDisabled: nextDisabledState }));
+    telemetry.setDisabled(nextDisabledState);
+  }, [state.telemetryDisabled, telemetry]);
+
+  return (
+    <View flex={1} backgroundColor="$background">
+      <ScreenHeader
+        backAction={() => props.navigation.goBack()}
+        title="Privacy Settings"
+      />
+      <View
+        flex={1}
+        width="100%"
+        maxWidth={600}
+        marginHorizontal="auto"
+        paddingHorizontal="$xl"
+      >
+        <YStack paddingHorizontal="$l" paddingTop="$2xl" gap="$xl">
+          <XStack justifyContent="space-between" alignItems="center">
+            <SizableText flexShrink={1}>Share Usage Statistics</SizableText>
+            <Switch
+              style={{ flexShrink: 0 }}
+              value={!state.telemetryDisabled}
```

Comment:
Seeing something weird with this locally (on Android). I install the app, login as self hosted, leave "share anonymous usage" disabled, then I come to this screen and it shows as enabled.

Reply:
My emulator is having trouble accessing localhost, so I haven't been able to test this on Android yet. Seems to work as expected on iOS.

That flag is set based on PostHog's opted out state, which is supposed to be updated when you navigate past the _SetTelemetry_ screen.

@patosullivan are you able to connect to local ships from your emulator? or were working with a remote self hosted one?

Reply:
this was an env var issue